src/AccessLogEntry.cc

```diff
@@ -88,3 +88,12 @@ AccessLogEntry::~AccessLogEntry()
 #endif
 }
 
+const SBuf *
+AccessLogEntry::effectiveVirginUrl() const
+{
+    const SBuf *effectiveUrl = request ? &request->url.absolute() : &virginUrlForMissingRequest_;
+    if (!effectiveUrl || effectiveUrl->isEmpty())
```

Comment:
Please revert this condition and return effectiveUrl from the `if` statement. It will make it easier to understand what kind of effectiveUrl we are returning (rather than reverse the condition in our mind). Besides, and "and" condition is usually easier to grasp than an "or" condition.